docs/template_plugin/tests/functional/op_reference/batch_to_space.cpp

```diff
@@ -0,0 +1,221 @@
+            Tensor({4}, element::i64, std::vector<int64_t>{1, 1, 2, 2}),
+            Tensor({4}, element::i64, std::vector<int64_t>{0, 0, 0, 0}),
+            Tensor({4}, element::i64, std::vector<int64_t>{0, 0, 0, 2}),
+            Tensor({1, 1, 4, 4}, IN_ET, std::vector<T>{1, 7,  2, 8,  13, 19, 14, 20,
+                                                       4, 10, 5, 11, 16, 22, 17, 23}),
+            "input_with_shape_with_crop_4x1x2x3"),
+        // input_with_shape_with_crop_4x1x2x3_1
+        BatchToSpaceParams(
+            Tensor({4, 1, 2, 3}, IN_ET, std::vector<T>{1,  2,  3,  4,  5,  6,
+                                                       7,  8,  9,  10, 11, 12,
+                                                       13, 14, 15, 16, 17, 18,
+                                                       19, 20, 21, 22, 23, 24}),
+            Tensor({4}, element::i64, std::vector<int64_t>{1, 1, 2, 2}),
+            Tensor({4}, element::i64, std::vector<int64_t>{0, 0, 0, 2}),
+            Tensor({4}, element::i64, std::vector<int64_t>{0, 0, 0, 0}),
+            Tensor({1, 1, 4, 4}, IN_ET, std::vector<T>{2, 8,  3, 9,  14, 20, 15, 21,
+                                                       5, 11, 6, 12, 17, 23, 18, 24}),
+            "input_with_shape_with_crop_4x1x2x3_1"),
+        // input_with_shape_with_crop_4x1x2x3_2
+        BatchToSpaceParams(
+            Tensor({4, 1, 2, 3}, IN_ET, std::vector<T>{1,  2,  3,  4,  5,  6,
+                                                       7,  8,  9,  10, 11, 12,
+                                                       13, 14, 15, 16, 17, 18,
+                                                       19, 20, 21, 22, 23, 24}),
+            Tensor({4}, element::i64, std::vector<int64_t>{1, 1, 2, 2}),
+            Tensor({4}, element::i64, std::vector<int64_t>{0, 0, 1, 0}),
+            Tensor({4}, element::i64, std::vector<int64_t>{0, 0, 1, 0}),
+            Tensor({1, 1, 2, 6}, IN_ET, std::vector<T>{13, 19, 14, 20, 15, 21,
+                                                       4,  10, 5,  11, 6,  12}),
+            "input_with_shape_with_crop_4x1x2x3_2"),
+    };
+    return batchToSpaceParams;
+}
+
+std::vector<BatchToSpaceParams> generateBatchToSpaceCombinedParams() {
+    const std::vector<std::vector<BatchToSpaceParams>> batchToSpaceTypeParams {
+        generateBatchToSpaceParams<element::Type_t::i8>(),
+        generateBatchToSpaceParams<element::Type_t::i16>(),
+        generateBatchToSpaceParams<element::Type_t::i32>(),
+        generateBatchToSpaceParams<element::Type_t::i64>(),
+        generateBatchToSpaceParams<element::Type_t::u8>(),
+        generateBatchToSpaceParams<element::Type_t::u16>(),
+        generateBatchToSpaceParams<element::Type_t::u32>(),
+        generateBatchToSpaceParams<element::Type_t::u64>(),
+        generateBatchToSpaceParams<element::Type_t::f16>(),
+        generateBatchToSpaceParams<element::Type_t::f32>(),
+        generateBatchToSpaceParams<element::Type_t::f64>(),
```

Comment:
In general we don't have any limitations for other data types, like `bf16`, `u4` and etc. in `BatchToSpace::evaluate()` method. Could we extend the set of supported data types?